alts/shared/uploaders/pulp.py

```diff
@@ -256,15 +256,15 @@ def upload_single_file(self, filename: str, type_: str = 'test_log') -> dict:
         """
         file_sha256 = hash_file(filename, hash_type='sha256')
         reference = self.check_if_artifact_exists(file_sha256)
-        if not reference:
-            reference = self._send_file(filename)
+        if reference:
+            raise UploadError((f"Unable to upload file with reference: {reference}"))
```

Comment:
maybe it`s better to change error name to make it more descriptive?
something like
`raise UploadError(f"File {filename} already exists:  href  {reference}")`

Also it looks a bit dangerous: does it mean that we will crash in case if someone tries to upload the same file again?

Reply:
Yes, we should crash if someone tries to upload the same file again. Because we can't save more than one file with the same reference (with the same content) and that's why we have 404 on logs sometimes